app/controllers/api/v1/rent_controller.rb

```diff
@@ -3,15 +3,15 @@ module V1
     class RentController < ApiController
       layout 'mailer'
       def index
-        if current_user
-          render_paginated Rent.where(user_id: current_user.id)
-        else
-          render json: 'Error: cannot find user'
+        Rent.where(user_id: current_user.id).each do |rent|
+          authorize rent
         end
+        render_paginated Rent.where(user_id: current_user.id)
       end
 
       def create
         rent = Rent.new(create_params)
+        authorize rent
```

Comment:
Ok, now we have a way to authorize endpoints. How did you manage the `:unauthorized` error? Maybe should be good using a `rescue_from` in your `ApiController` to handle the error?